core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -1399,6 +1397,158 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
+    topicConfigEntries1 = Seq(
+      new AlterConfigOp(new ConfigEntry(LogConfig.CompressionTypeProp, "zip"), AlterConfigOp.OpType.SET)
+    ).asJava
+
+    alterResult = client.incrementalAlterConfigs(Map(
+      topicResource1 -> topicConfigEntries1
+    ).asJava, new IncrementalAlterConfigsOptions().validateOnly(true))
+
+    assertFutureExceptionTypeEquals(alterResult.values().get(topicResource1), classOf[InvalidRequestException],
+      Some("Invalid config value for resource"))
+  }
+
+  @Test
+  def testInValidIncrementalAlterConfigs(): Unit = {
+    client = AdminClient.create(createConfig)
+
+    // Create topics
+    val topic1 = "incremental-alter-configs-topic-1"
+    val topicResource1 = new ConfigResource(ConfigResource.Type.TOPIC, topic1)
+    createTopic(topic1)
+
+    val topic2 = "incremental-alter-configs-topic-2"
+    val topicResource2 = new ConfigResource(ConfigResource.Type.TOPIC, topic2)
+    createTopic(topic2)
+
+    var topicConfigEntries1 = Seq(
+      new AlterConfigOp(new ConfigEntry(LogConfig.CleanupPolicyProp, LogConfig.Delete), AlterConfigOp.OpType.APPEND),
+      new AlterConfigOp(new ConfigEntry(LogConfig.CleanupPolicyProp, LogConfig.Compact), AlterConfigOp.OpType.DELETE)
+    ).asJavaCollection
+
+    var topicConfigEntries2 = Seq(
+      new AlterConfigOp(new ConfigEntry(LogConfig.MinCleanableDirtyRatioProp, "0.9"), AlterConfigOp.OpType.SET)
+    ).asJavaCollection
+
+    var alterResult = client.incrementalAlterConfigs(Map(
+      topicResource1 -> topicConfigEntries1,
+      topicResource2 -> topicConfigEntries2
+    ).asJava)
+    assertEquals(Set(topicResource1, topicResource2).asJava, alterResult.values.keySet)
+    //valid result
+    alterResult.values().get(topicResource2).get()
+
+    assertFutureExceptionTypeEquals(alterResult.values().get(topicResource1), classOf[InvalidRequestException],
+      Some("Duplicate config keys provided for resource"))
+
+
+    //invalid use of append/substract operation type
+    topicConfigEntries1 = Seq(
+      new AlterConfigOp(new ConfigEntry(LogConfig.CompressionTypeProp, "gzip"), AlterConfigOp.OpType.APPEND)
+    ).asJavaCollection
+
+    topicConfigEntries2 = Seq(
+      new AlterConfigOp(new ConfigEntry(LogConfig.CompressionTypeProp, "snappy"), AlterConfigOp.OpType.SUBSTRACT)
+    ).asJavaCollection
+
+    alterResult = client.incrementalAlterConfigs(Map(
+      topicResource1 -> topicConfigEntries1,
+      topicResource2 -> topicConfigEntries2
+    ).asJava)
+    assertEquals(Set(topicResource1, topicResource2).asJava, alterResult.values.keySet)
+
+    assertFutureExceptionTypeEquals(alterResult.values().get(topicResource1), classOf[InvalidRequestException],
+      Some("Config value append is not allowed for config"))
+
+    assertFutureExceptionTypeEquals(alterResult.values().get(topicResource2), classOf[InvalidRequestException],
+      Some("Config value substract is not allowed for config"))
+  }
```

Comment:
So far we've only tested against IncrementalAlterConfig semantics. What do you think about adding a quick assert about an invalid property value to ensure those get validated as well?  This can be equivalent to the 
```java
      new ConfigEntry(LogConfig.MinCleanableDirtyRatioProp, "1.1"), // this value is invalid as it's above 1.0
```
check made in `def checkInvalidAlterConfigs`

Reply:
updated the test